src/Classes/Utils/Utils.ts

```diff
@@ -84,48 +126,64 @@ export class Utils {
         return velocityBase;
     }
 
-    public static getParticleColor(options: IOptions, color: { value: string[] | IColor | string }): IColor {
-        const res: IColor = {};
+    public static getParticleColor(options: IOptions, color: { value: string[] | IColor | string }): IRgb | null {
+        let res: IRgb | null = null;
 
         if (typeof (color.value) === "object") {
             if (color.value instanceof Array) {
                 const arr = options.particles.color.value as string[];
                 const colorSelected = color.value[Math.floor(Math.random() * arr.length)];
 
-                res.rgb = Utils.hexToRgb(colorSelected);
+                res = Utils.hexToRgb(colorSelected);
             } else {
                 const rgbColor = color.value as IRgb;
 
                 if (rgbColor.r !== undefined) {
-                    res.rgb = {
-                        b: rgbColor.b,
-                        g: rgbColor.g,
-                        r: rgbColor.r,
-                    };
+                    res = rgbColor;
                 }
 
                 const hslColor = color.value as IHsl;
 
                 if (hslColor.h !== undefined) {
-                    res.hsl = {
-                        h: hslColor.h,
-                        l: hslColor.l,
-                        s: hslColor.s,
-                    };
+                    res = Utils.hslToRgb(hslColor);
                 }
             }
         } else {
             if (color.value === "random") {
-                res.rgb = {
+                res = {
                     b: Math.floor(Math.random() * 256),
                     g: Math.floor(Math.random() * 256),
                     r: Math.floor(Math.random() * 256),
                 };
             } else {
-                res.rgb = Utils.hexToRgb(color.value);
+                res = Utils.hexToRgb(color.value);
             }
         }
 
         return res;
     }
+
+    private static hue2rgb(p: number, q: number, t: number): number {
+        if (t < 0) {
+            t += 1;
+        }
+
+        if (t > 1) {
+            t -= 1;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Reassigning parameter 't' is forbidden.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5098130)